File: from-3601-to-3700/3622-check-divisibility-by-digit-sum-and-product/solution.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        num = n
        dSumm, dProd = 0, 1
        while n > 0:
            dig = n % 10
            dSumm += dig
            dProd *= dig
            n //= 10

        return num % (dSumm + dProd) == 0
        

File: from-3601-to-3700/3622-check-divisibility-by-digit-sum-and-product/solution.go
func checkDivisibility(n int) bool {
    num := n
    dSumm, dProd := 0, 1
    for n > 0 {
        dig := n % 10
        dSumm += dig
        dProd *= dig
        n /= 10
    }
    return num % (dSumm + dProd) == 0
}